contracts/kernel/Kernel.sol

```diff
@@ -47,52 +41,44 @@ contract Kernel is IKernel, KernelStorage, Initializable {
     *      if a created permission is removed it is possible to reset it with createPermission.
-    * @notice Create a new permission granting `_entity` the ability to perform actions of role `_role` on `_app` (setting `_parent` as parent)
+    * @notice Create a new permission granting `_entity` the ability to perform actions of role `_role` on `_app` (setting `_owner` as parent)
     * @param _entity Address of the whitelisted entity that will be able to perform the role
     * @param _app Address of the app in which the role will be allowed (requires app to depend on kernel for ACL)
     * @param _role Identifier for the group of actions in app given access to perform
-    * @param _parent Address of the entity that will be able to revoke the permission. If set to `_entity`, then it will be able to grant it too
+    * @param _owner Address of the entity that will be able to grant and revoke the permission further.
     */
     function createPermission(
         address _entity,
         address _app,
         bytes32 _role,
-        address _parent
-    ) auth(CREATE_PERMISSIONS_ROLE) external
+        address _owner
+    )   auth(CREATE_PERMISSIONS_ROLE)
+        external
     {
         _createPermission(
             _entity,
             _app,
             _role,
-            _parent
+            _owner
         );
     }
 
     /**
-    * @dev Grants a permission if allowed. This requires `msg.sender` to have the permission and be its own parent
-    * @notice Grants `_entity` the ability to perform actions of role `_role` on `_app` (setting `_parent` as parent)
+    * @dev Grants a permission if allowed. This requires `msg.sender` to be the permission owner
+    * @notice Grants `_entity` the ability to perform actions of role `_role` on `_app`
     * @param _entity Address of the whitelisted entity that will be able to perform the role
     * @param _app Address of the app in which the role will be allowed (requires app to depend on kernel for ACL)
     * @param _role Identifier for the group of actions in app given access to perform
-    * @param _parent Address of the entity that will be able to revoke the permission. If set to `_entity`, then it will be able to grant it too
     */
-    function grantPermission(
-        address _entity,
-        address _app,
-        bytes32 _role,
-        address _parent
-    ) external
+    function grantPermission(address _entity, address _app, bytes32 _role)
+             onlyPermissionOwner(_app, _role)
+             external
     {
-        // Implicitely check parent has permission and its parent of it (if it didn't have permission, it wouldn't have a parent)
-        require(permissions[msg.sender][_app][_role].parent == msg.sender);
-        // Permission can only can be set if entity doesn't already have it
-        require(permissions[_entity][_app][_role].allowed == false);
-
+        // Only permission owner can grant permission
```

Comment:
Maybe move this comment up to the line with the `onlyPermissionOwner` modifier?

Reply:
Removing it all together as the modifier is self explanatory